src/gemm.rs

```diff
@@ -115,6 +115,20 @@ unsafe fn gemm_loop<K>(
     where K: GemmKernel
 {
     debug_assert!(m * n == 0 || (rsc != 0 && csc != 0));
+    if m == 0 || k == 0 || n == 0 {
```

Comment:
Thanks for finding this! I think this should be just "if k == 0" ? That's the case where the C matrix can have more than 0 elements. And a comment, `// if A or B have no elements, compute C ← βC and return`

Benchmarks pass for me with this PR, so it should be fine as it is.

Reply:
I added the comment.

Yeah, the `m == 0` and `n == 0` conditions aren't necessary, but I added them for a few reasons:

1. The rest of the implementation no longer has to worry about `m` or `n` being zero.
2. For people reviewing the code, it's easy to see that this block works correctly for `m == 0` and `n == 0`, while it's less obvious that's true for the rest of the implementation.
3. For these cases, it's nice to finish early instead of continuing to the rest of the implementation.

I can remove `m == 0` and `n == 0` if you'd prefer, though.